hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AuthenticationBaseMessageTask.java

```diff
@@ -95,6 +100,11 @@ private ClientMessage handleAuthenticated() {
             }
         }
 
+        if (clientEngine.getClusterService().getMember(principal.getOwnerUuid()) == null) {
```

Comment:
I try to prevent 'complex' expression in if statements because it isn't obvious what you are doing.

```
boolean isMember = clientEngine.getClusterService().getMember(principal.getOwnerUuid());
if(!isMember){
}
```

This is a lot easier to understand since it explain what you are doing.